packages/widgets/src/menubar.ts

```diff
@@ -403,8 +457,72 @@ export class MenuBar extends Widget {
           this.activeIndex = i;
         }
       });
+      // Calculate size of current menu
+      totalMenuSize += this._menuItemSizes[i];
+    }
+    // Render overflow menu if needed
+    if (this._overflowIndex > -1) {
+      // Create overflow menu
+      if (this._overflowMenu === null) {
+        this._overflowMenu = new Menu({ commands: new CommandRegistry() });
+        this._overflowMenu.title.label = '...';
+        this._overflowMenu.title.mnemonic = 0;
+        this.addMenu(this._overflowMenu, false);
+      }
+      // Move menus to overflow menu
+      for (let i = menus.length - 2; i >= length; i--) {
+        let submenu = this.menus[i];
+        submenu.title.mnemonic = 0;
+        this._overflowMenu.insertItem(0, {
+          type: 'submenu',
+          submenu: submenu
+        });
+        this.removeMenu(submenu, false);
+      }
+      let title = this._overflowMenu.title;
+      let active = length === activeIndex;
+      if (active && menus[length].items.length == 0) {
+        active = false;
+      }
+      content[length] = renderer.renderItem({
+        title,
+        active,
+        onfocus: () => {
+          this.activeIndex = length;
+        }
+      });
+    } else if (this._overflowMenu !== null) {
+      // Remove submenus from overflow menu
+      let overflowMenuItems = this._overflowMenu.items;
+      let screenSize = this.node.offsetWidth;
+      let n = this._overflowMenu.items.length;
+      for (let i = 0; i < n; ++i) {
+        let index = menus.length - 1 - i;
+        if (screenSize - totalMenuSize > this._menuItemSizes[index]) {
+          let menu = overflowMenuItems[0].submenu as Menu;
+          this._overflowMenu.removeItemAt(0);
+          this.insertMenu(length, menu, false);
+          let title = menu.title;
+          let active = false;
+          content[length] = renderer.renderItem({
+            title,
+            active,
```

Comment:
Let's fold it to reduce mental burden.

```suggestion
          content[length] = renderer.renderItem({
            title: menu.title,
            active: false,
```

Reply:
I get lint errors when I do it this way... not sure why

Reply:
What is the error?

Reply:
```
Argument of type '{ this: any; active: boolean; onfocus: () => void; }' is not assignable to parameter of type 'IRenderData'.
  Object literal may only specify known properties, and 'this' does not exist in type 'IRenderData'.ts(2345)
```

Reply:
I do not see this error locally. Should we see if it is reproducible on CI?

Reply:
let's try it!